docs/book/src/developer/providers/v1.2-to-v1.3.md

```diff
@@ -0,0 +1,37 @@
+# Cluster API v1.2 compared to v1.3
```

Comment:
Let's drop this file.

the release-1.2 branch is for v1.2.x so there is a v1.1=>v1.2 file instead. But there shouldn't be any changes we have to make in this PR